observatory-dags/observatory/dags/telescopes/grid.py

```diff
@@ -1,216 +1,224 @@
-                jsonl_gzip_file.write(bytes_io.getvalue())
+            # Parse date:
+            # The publish date is not used as the release date because the dataset is often
+            # published after the release date
+            date_matches = re.search("([0-9]{4}\-[0-9]{2}\-[0-9]{2})", title)
+            if date_matches is None:
+                raise ValueError(f'No release date found in GRID title: {title}')
+            release_date = pendulum.parse(date_matches[0])
 
-    return version, file_name, file_path
+            try:
+                release_articles[release_date].append(article_id)
+            except KeyError:
+                release_articles[release_date] = [article_id]
 
+    for release_date in release_articles:
+        article_ids = release_articles[release_date]
+        records.append({
+            'article_ids': article_ids,
+            'release_date': release_date
+        })
+    return records
 
-class GridTelescope:
-    """ A container for holding the constants and static functions for the GRID telescope. """
 
+class GridTelescope(SnapshotTelescope):
+    """ The Global Research Identifier Database (GRID): https://grid.ac/ """
     DAG_ID = 'grid'
-    DATASET_ID = 'digital_science'
-    DESCRIPTION = 'The Global Research Identifier Database (GRID): https://grid.ac/'
-    RELEASES_TOPIC_NAME = 'releases'
-    QUEUE = 'default'
-    RETRIES = 3
-
-    TASK_ID_CHECK_DEPENDENCIES = 'check_dependencies'
-    TASK_ID_LIST = 'list_releases'
-    TASK_ID_DOWNLOAD = 'download'
-    TASK_ID_UPLOAD_DOWNLOADED = 'upload_downloaded'
-    TASK_ID_EXTRACT = 'extract'
-    TASK_ID_TRANSFORM = 'transform'
-    TASK_ID_UPLOAD_TRANSFORMED = 'upload_transformed'
-    TASK_ID_BQ_LOAD = 'bq_load'
-    TASK_ID_CLEANUP = 'cleanup'
-    TASK_ID_STOP = 'stop_dag'
-
-    @staticmethod
-    def check_dependencies(**kwargs):
-        """ Check that all variables exist that are required to run the DAG.
-
-        :param kwargs: the context passed from the PythonOperator. See
-        https://airflow.apache.org/docs/stable/macros-ref.html
-        for a list of the keyword arguments that are passed to this argument.
-        :return: None.
-        """
-
-        vars_valid = check_variables(AirflowVars.DATA_PATH, AirflowVars.PROJECT_ID,
-                                     AirflowVars.DATA_LOCATION, AirflowVars.DOWNLOAD_BUCKET,
-                                     AirflowVars.TRANSFORM_BUCKET)
-        if not vars_valid:
-            raise AirflowException('Required variables are missing')
-
-    @staticmethod
-    def list_releases(**kwargs):
-        """ Task to lists all GRID releases for a given month.
+    GRID_FILE_URL = "https://api.figshare.com/v2/articles/{article_id}/files"
+    GRID_DATASET_URL = "https://api.figshare.com/v2/collections/3812929/articles?page_size=1000"
+
+    def __init__(self):
+        self.dag_id = GridTelescope.DAG_ID
+        self.start_date = datetime(2015, 9, 1)
```

Comment:
I'm not too sure. I mentioned it briefly in the google doc (sorry I did not point it out explicitly that I updated this), here: https://docs.google.com/document/d/1Tpk9Fq-6hrwST9JD7HtHOT2UWhzD9bJfDZp-jD3k7co/edit#bookmark=id.tt3r0aq8x7n8

I think it would be nice to have the parameters in one location, so you can check the values in one glance. To me it makes most sense to do it in the telescope specific file (like here telescopes/grid.py), but perhaps it should be done when initialising the GridTelescope in dags/grid.py.